Buildings/Controls/OBC/Utilities/PIDWithInputGains.mo

```diff
@@ -0,0 +1,749 @@
+    "Typical range of control error, used for scaling the control error";
+  parameter Real yMax=1
+    "Upper limit of output"
+    annotation (Dialog(group="Limits"));
+  parameter Real yMin=0
+    "Lower limit of output"
+    annotation (Dialog(group="Limits"));
+  parameter Real Ni(
+    min=100*CDL.Constants.eps)=0.9
+    "Ni*Ti is time constant of anti-windup compensation"
+    annotation (Dialog(tab="Advanced",group="Integrator anti-windup",enable=controllerType == CDL.Types.SimpleController.PI or controllerType ==CDL.Types.SimpleController.PID));
+  parameter Real Nd(
+    min=100*CDL.Constants.eps)=10
+    "The higher Nd, the more ideal the derivative block"
+    annotation (Dialog(tab="Advanced",group="Derivative block",enable=controllerType == CDL.Types.SimpleController.PD or controllerType ==CDL.Types.SimpleController.PID));
+  parameter Real xi_start=0
+    "Initial value of integrator state"
+    annotation (Dialog(tab="Advanced",group="Initialization",enable=controllerType == CDL.Types.SimpleController.PI or controllerType == CDL.Types.SimpleController.PID));
+  parameter Real yd_start=0
+    "Initial value of derivative output"
+    annotation (Dialog(tab="Advanced",group="Initialization",enable=controllerType == CDL.Types.SimpleController.PD or controllerType == CDL.Types.SimpleController.PID));
+  parameter Boolean reverseActing=true
+    "Set to true for reverse acting, or false for direct acting control action";
+  parameter Real y_reset=xi_start
+    "Value to which the controller output is reset if the boolean trigger has a rising edge"
+    annotation (Dialog(enable=controllerType == CDL.Types.SimpleController.PI or controllerType == CDL.Types.SimpleController.PID,group="Integrator reset"));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_s
+    "Connector of setpoint input signal"
+    annotation (Placement(transformation(extent={{-260,-20},{-220,20}}),iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_m
+    "Connector of measurement input signal"
+    annotation (Placement(transformation(origin={0,-220},extent={{20,-20},{-20,20}},rotation=270),iconTransformation(extent={{20,-20},{-20,20}},rotation=270,origin={0,-120})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput k
+    "Connector of control gain signal"
+    annotation (Placement(transformation(extent={{-260,120},{-220,160}}),iconTransformation(extent={{-140,60},{-100,100}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput Ti if with_I
+    "Connector of time constant signal for the integral term"
+    annotation (Placement(transformation(extent={{-260,60},{-220,100}}),iconTransformation(extent={{-140,20},{-100,60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput Td if with_D
+    "Connector of time constant signal for the derivative term"
+    annotation (Placement(transformation(extent={{-260,-100},{-220,-60}}),iconTransformation(extent={{-140,-60},{-100,-20}})));
```

Comment:
In the graphical implementation, it would be better if you first list all gains, and then the input `u_m`, or vice versa. Currently, two gains are listed, then the input `u_m`, and then another gain, which does not look clean.

Moreover, `CDL.Continuous.PIDWithReset`, the parameters `Ti` and `Td` have units and quantities, so the corresponding inputs must also have units and quantities. Also, the `min` attribute need to be added, similar as is used in the parameters of `PIDWithReset`.

Reply:
Thanks.
Not sure if I follow you. The current order of the input is "u_i, u_m, k, Ti, Td". Can you let me know if this is desired?
I have added units, quantities, and min to <code>k</code>, <code>Ti</code>, and <code>Td</code>.